### DISCUSSION TITLE
master/detail view or FileMaker portal equivalents?

### DISCUSSION THREAD

**Discussion by tvogt** [Budibase/budibase#7440] [Help]
I'm currently evaluating Budibase so I have very little experience with it. The closest to it that I'm familiar with is FileMaker.

I know that I will be needing the following and can't figure out how to do it or if it is possible at all:

1. I have a table of questions. Mostly just a title and a text. They are grouped into categories. The categories can be an option field or another table. 
2. I have a table of people.
3. I have an answers (junction) table that links these two. It contains question (reference), person (reference) and answer (text field).

I'd like to have an interface similar to readthedocs, with all the questions of a particular category on the left sidebar, and when I click on it the right the full question opens and I can enter an answer. The person that I'm interviewing would be selected beforehand, I don't particularily care how. I assume their ID would be in the URL then.

1. How do I filter the questions on the left to show only the ones in the same category as the currently selected one? I do have URL.id but can't figure out how to get data from the current entry.
2. How do I show the data from the related table? My url ID would be the answer but I want to also show the text from the related question on the page.

1. On the screen for people have a button that creates a new set of empty answers (junction table entries) for a particular category only. Basically "start up interview X for person A".
4. Then I'd like to have a screen with all the questions for a selected person, and I can click on the question to open the answer so I can input it. Think readthedocs for interface - questions on the left, click on them, then enter details on the right, save when clicking on the next question (and/or a save button).

I've been trying for a few hours and just can't get the relations up, or the data providers to fetch the right data set, or anything. I found a few similar questions, tried their answers but they are sufficiently different that they didn't work for me. I'm probably missing something like a principle of how forms and data providers and filters work.

edit: I think the more general question here is: When I am on a design with data from table A, how do I access/display data from related tables?

**Comment by melohagan** [comment]
Hi @tvogt 

Let me have a go at answering your questions.

1. There are some docs on filtering a data provider [here](https://docs.budibase.com/docs/searchfilter-data#static-filtering). The video at the bottom explains filtering via options picker e.g. your chosen category. 
2. [Displaying data from relatonships](https://docs.budibase.com/docs/fetching-data#displaying-relational-data)
3. If you want to create empty answers for a particular category, then you may want your categories to also be in a table. The category could be chosen from a [Relationship Picker](https://docs.budibase.com/docs/pickers#relationship-picker) and then your could save a new *Answer* with a relationship to the chosen category relationship.
4. Saving to multiple tables in one form may help here: https://docs.budibase.com/docs/referencing#example-creating-and-linking-two-table-entries-in-one-form

There's quite a bit to take in there, so if you want to hone in on one specific task at a time then I'm happy to help you out with that.
